src/Microsoft.AspNetCore.Hosting/Internal/RequestServicesFeature.cs

```diff
@@ -9,19 +9,19 @@ namespace Microsoft.AspNetCore.Hosting.Internal
 {
     public class RequestServicesFeature : IServiceProvidersFeature, IDisposable
     {
-        private IServiceProvider _appServices;
+        private RequestServicesContainerMiddleware _middleware;
         private IServiceProvider _requestServices;
         private IServiceScope _scope;
         private bool _requestServicesSet;
 
-        public RequestServicesFeature(IServiceProvider applicationServices)
+        public RequestServicesFeature(RequestServicesContainerMiddleware middleware)
```

Comment:
Thought about passing a `Func<IServiceScopeFactory>` here but we already have the middleware object so we allocate another :smile: